packages/babel-plugin-async-load/src/index.js

```diff
@@ -0,0 +1,58 @@
+const lazyLoadTemplate = `
+var COMPONENT_NAME = wp.compose.lazyLoad( SCRIPTS, STYLES, BASE_URL)( WRAPPED_COMPONENT );
+`;
+
+export default function( babel ) {
+	const { types: t, template } = babel;
+
+	return {
+		visitor: {
+			Program: function( path, state ) {
+				// 1. Find occurences to replace
+				const mappings = [];
+				const importVisitor = {
+					ImportSpecifier: function( subPath ) {
+						state.opts.components.forEach( ( componentConfig ) => {
+							if (
+								subPath.parent.source.value === componentConfig.module &&
+								subPath.node.imported.name === componentConfig.component
+							) {
+								// Add the Wrapper
+								const buildRequire = template( lazyLoadTemplate );
+								const ast = buildRequire( {
+									WRAPPED_COMPONENT: t.identifier( subPath.node.local.name ),
+									COMPONENT_NAME: t.identifier( 'LazyLoad' + subPath.node.local.name ),
+									SCRIPTS: t.arrayExpression( componentConfig.scripts.map( ( script ) => t.stringLiteral( script ) ) ),
+									STYLES: t.arrayExpression( componentConfig.styles.map( ( style ) => t.stringLiteral( style ) ) ),
+									BASE_URL: t.identifier( state.opts.siteURLSource ),
```

Comment:
Initial reaction: Why does a Babel plugin care about a WordPress site URL?

_Edit:_ Second reaction: Can't we just allow this to be defined as some generic option which is serialized and passed through to the base component via the template call?

_Edit:_ Third reaction: Okay, duh, I understand now; the loading is satisfied by calling `load-scripts.php`. I still wonder if, as part of this being a generic non-WordPress-specific Babel plugin, we could change this to be satisfied by an implementation provided by the consumer. Maybe a global function which returns a promise when called, that the `lazyLoad` wrapper will execute (and pass serialized options for specific components, that we could use to pass the script handles?)